lib/cocoapods/target/build_settings.rb

```diff
@@ -555,12 +555,11 @@ def initialize(target, non_library_spec = nil)
 
           frameworks = []
           frameworks.concat consumer_frameworks
-          if non_library_xcconfig? || (target.should_build? && target.build_as_dynamic?)
+          if library_xcconfig? && (target.should_build? && target.build_as_dynamic?)
```

Comment:
this is an interesting and scary changes? can you clarify why? I anticipate it will break many tests

Reply:
breaks 0 tests, what was happening is we were adding these frameworks twice before for the non_library case, once in this conditional and the other from the dependent targets